### ISSUE TITLE
[FEATURE] Support fuse for Fileset

### ISSUE THREAD

**Issue by coolderli** [apache/gravitino#4558]
### Describe the feature

Implement fuse for gvfs to support mounting fileset to local directories. The instance defaults to mounting `fileset://fileset/fileset_catalog/schema/fileset_name` to `/fileset/fileset_catalog/schema/fileset_name`. So we can access it via posix protocol. In addition, we can support mounting to user-defined directories, so that users do not need to modify any code.

### Motivation

In AI scenarios, users often use the posix protocol to access data. The data is stored in media such as JuiceFS, NAS, or CPFS, and then mounted to a local directory.
Directly using these storage has the following disadvantages:
1. Authentication issues: Each product has its authentication system, and users have a high access cost
2. Authorization issues: When users share data, they need to share the permissions of the entire volume, and there is no way to perform fine-grained permission control
3. Audit issues: Some NAS products lack audit logs, making it impossible to perform effective audits and data deletion

### Describe the solution

1. Use the underlying fuse directly.  
This means that fileset needs to manage a local directory. I think this is not a good solution, users will bypass gvfs without any benefits.

2. Using fsspec fuse to implement gvfs fuse
fsspec provides the feature of fuse, which supports forwarding fuse operations to fsspec fs operations: https://filesystem-spec.readthedocs.io/en/latest/features.html#mount-anything-with-fuse
We could do some optimization based on fsspec fuse to support gvfs fuse.

3. Implement gvfs fuse using JNI to call GravitinoVirtualFileSystem
![image](https://github.com/user-attachments/assets/ce303bdc-623a-4845-9b5b-d6f8d56ff06a)

At present, Solution 2 and Solution 3 are similar. Solution 2 is implemented by calling Python gvfs, and Solution 3 is implemented by calling Java gvfs.

**Comment by coolderli** [comment]
@jerryshao @shaofengshi @xunliu Can you share your thoughts？Thanks. cc @xloya @YxAc @zhoukangcn

**Comment by jerryshao** [comment]
Thanks @coolderli to bring this out. I think HCFS API is not Posix compliant API, so using HCFS API with fuse has many limitations. I don't know how well fsspec support Posix, we need to investigate.

The concern from my side is the performance of using python to achieve fuse, using fuse requires lots of context switch (between user space and kernel space) and it will affect the performance a lot. If we use a dynamic language, then the performance will be worse.

Currently, we don't have a better solution, maybe we should investigate more to have a better solution.

**Comment by shaofengshi** [comment]
Hi Peidian, I don't have much knowledge about FUSE; For the solution2, is that only available for python? Which means, only in Python application, fsspec mounts a remote storage as a local path, and the user can read/write that from Python codes.

**Comment by coolderli** [comment]
> Hi Peidian, I don't have much knowledge about FUSE; For the solution2, is that only available for python? Which means, only in Python application, fsspec mounts a remote storage as a local path, and the user can read/write that from Python codes.

@shaofengshi Not exactly. Users need to run a piece of Python code to perform a mount operation first, and then they can use other applications to access it.

**Comment by shaofengshi** [comment]
> > Hi Peidian, I don't have much knowledge about FUSE; For the solution2, is that only available for python? Which means, only in Python application, fsspec mounts a remote storage as a local path, and the user can read/write that from Python codes.
> 
> @shaofengshi Not exactly. Users need to run a piece of Python code to perform a mount operation first, and then they can use other applications to access it.

Got it; thanks Peidian's input; Not sure how compatible and stable it is, such as OS support, Python versions etc. If that is not very good, we may not be able to persuade a large group of user to use it.

**Comment by diqiu50** [comment]
Hi @coolderli  I heard that you encountered some issues with `fsspec`. I’d like to understand the details. Could you help list the specific problems?

**Comment by coolderli** [comment]
> Hi @coolderli I heard that you encountered some issues with `fsspec`. I’d like to understand the details. Could you help list the specific problems?

The first issue is from [fusepy](https://github.com/fusepy/fusepy/blob/master/fuse.py#L587). It's about TypeError, meaning datetime can't be converted to an int. I have fixed it.
![image](https://github.com/user-attachments/assets/1e476f83-3151-4851-81f9-c084f73f412f)

For now, the fsspec fuse can work. I submit a draft PR about fsspec fuse. You can take a look and have a try. And I need more tests.
https://github.com/apache/gravitino/pull/4634

**Comment by coolderli** [comment]
@diqiu50 I'm not sure if the fsspec fuse is a good way. Because in most cases we need to use it in a container environment, we not only need to implement fuse, but also need to implement k8s CSI. https://juicefs.com/docs/zh/csi/introduction/

On the other hand, using CSI in a cloud environment may not be feasible because it requires some configuration in the k8s cluster, although this is not a problem in our private cluster. Using fsspec fuse can help us avoid this situation, as we do not need to make any modifications to the existing k8s cluster. You can take a look at this article: https://mp.weixin.qq.com/s/j6AlSqKxKInAKeBfADJdOA.  The Juice team has also raised similar concerns about CSI.

**Comment by diqiu50** [comment]
@coolderli Does the problem occur only when using the HDFS file system?

**Comment by diqiu50** [comment]
We track the development process through issue #5504
